applications/OptimizationApplication/custom_utilities/filtering/explicit_filter_utils.cpp

```diff
@@ -101,82 +80,75 @@ double GetDomainSize(
     return pExpression->Evaluate(rPoint.Id(), rPoint.Id(), 0);
 }
 
-struct NonIntegratedWeight
+struct MeshIndependentType
 {
     static double Compute(
-        const double FilterFunctionWeight,
+        const double Value,
         const double DomainSize)
     {
-        return FilterFunctionWeight;
+        return Value;
     }
 };
 
-struct IntegratedWeight
+struct MeshDependentType
 {
     static double Compute(
-        const double FilterFunctionWeight,
+        const double Value,
         const double DomainSize)
     {
-        return FilterFunctionWeight * DomainSize;
+        return Value / DomainSize;
     }
 };
 
-template <class TEntityType, class TWeightComputationType>
+template <class TEntityType>
 void ComputeWeightForAllNeighbors(
```

Comment:
as i said in the prev PR, ween to stick to the original impl, that's why in your results you get bad tranbsition along damoing boundaries. Please take the original master implementation of this function. This is very crucuial !

Reply:
Usually damping is applied as a diagonal matrix $`\mathbf{D}`$, and filtering is done using $`\mathbf{A}`$.
```math
\underline{\Delta s} = \mathbf{D}\mathbf{A}\underline{\Delta p}
```

So, the master implementation combines these two implementations and does the following:
```math
\underline{\Delta s} = \mathbf{\tilde{A}}\underline{\Delta p}
```

Then we won't be able to implement different damping matrices to have modularity. The boundary chekerboard problem will be solved in a future PR with a different damping matrix mechanism (@Igarizza).

I would like to keep the damping matrix seperate and filtering matrix seperate in the explicit case, because it can be seperated easily.

As far as I see, mathematically also they should be seperate matrices, so if we can do it, I would like allow it.

Reply:
@sunethwarna This is a delicate point since in the master implementation damping weights are properly blended into the filtering operation so that we get a nice and smooth transition along fixed boundaries, that's why you got edgy features close to the fixed boundaries. I will not kill the feature since it is highly demanded and this property should be preserved.

Reply:
After discussion with @sunethwarna, we have decided to have both implementations co-exist because there is no easy way to have VM with integrated damping as a separate filtering method within the current implementation, because you have a nicely implemented solution for smoothed damping field which is not a diagonal matrix anymore. @RezaNajian How do you want to name it? In future, we plan to implement @DSchmoelz 's version of robust damping as well.

Reply:
Hi @RezaNajian, Your implementation is reverted back. Now both of the filter implementations are there because they do filtering in different ways. We want to have a conservative filter for explicit as well because we are planning to use it on point cloud filtering for SI. 

The `explicit_filter` will be the most generic one which will support as @Igarizza mentioned, different damping methods as well in the future. I used the name "explicit_smooth_damping_filter" for your implementation since it is a specific damping case. Let me know what you think?

Reply:
Ok, finally, I have integrated both damping mechanisms and implemented tests and put a nice gif as well. @Igarizza @RezaNajian .

Finally I uderstood what you were doing in @RezaNajian 's imlemenation. Now we have all of them with conservative mapping, and modular so others can implement their own sofisticated damping. :)